### PR TITLE
Add support for Chocolatey (#1612)

### PR DESCRIPTION
**Pull request by uilianries**
Hi!

This PR brings  Chocolatey into `SystemPackageTool`. 

Some points may be observed:
* Chocolatey doesn't have a real `update` like *apt* and *brew*, but there is an opened [issue](https://github.com/chocolatey/chocolatey/issues/320). For such feature, I'm used `choco outdated`, a shortcut for `choco upgrade all --noop`, to update information about installed packages.
* When Chocolatey search for local packages, it returns *%ERROLEVEL% == 0* when a package is installed or not. I used *findstr* to validate if such package is installed.
* The original package test expects *git* installed on development machine. By default, is not installed on AppVeyor, so I used *chocolatey* package.

Related issue: #1612 

Saludos.

### CODE REVIEW

**Comment by uilianries** [comment]
After this PR, I can update on Conan docs.

File: conans/test/util/tools_test.py [hunk 1]
```diff
@@ -124,9 +124,9 @@ def test_environment_nested(self):
     def system_package_tool_fail_when_not_0_returned_test(self):
         runner = RunnerMock(return_ok=False)
         spt = SystemPackageTool(runner=runner)
-        if platform.system() == "Linux" or platform.system() == "Darwin":
-            msg = "Command 'sudo apt-get update' failed" if platform.system() == "Linux" \
-                else "Command 'brew update' failed"
+        platforms = {"Linux": "sudo apt-get update", "Darwin": "brew update", "Windows": "choco outdated"}
+        if platform.system() in platforms.keys():
```
Comment (lasote) [inline conans/test/util/tools_test.py]: `.keys()` not needed.

Comment (uilianries) [inline conans/test/util/tools_test.py]: fixed!

**Comment by memsharded** [comment]
Perfect, many thanks!

**Pull request merged**